cpp/src/arrow/compute/kernels/common.h

```diff
@@ -19,6 +19,7 @@
 
 // IWYU pragma: begin_exports
 
+#include <cmath>
```

Comment:
Why add this to `common.h`?

Reply:
Initially I included cmath.h to use std::isnan(). I thought this header would be quite common to most kernels... my assumption might be wrong though ?

Reply:
We only add inclusion in header files when it is required by the header file itself, otherwise it would inflate compilation times for nothing. Admittedly, `cmath` is probably not a large file.

Reply:
Good point. Moving the include in scalar_validity.cc.